src/Razor/src/Microsoft.NET.Sdk.Razor/StaticWebAssetsGeneratePackagePropsFile.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class StaticWebAssetsGeneratePackagePropsFile : Task
+    {
+        [Required]
+        public string PropsFileImport { get; set; }
+
+        [Required]
+        public string BuildTargetPath { get; set; }
+
+        public override bool Execute()
+        {
+            var document = new XDocument(new XDeclaration("1.0", "utf-8", "yes"));
+            var root = new XElement(
+                "Project",
+                new XElement("Import",
+                    new XAttribute("Project", PropsFileImport)));
+
+            document.Add(root);
+
+            var settings = new XmlWriterSettings
+            {
+                Encoding = Encoding.UTF8,
+                CloseOutput = true,
+                OmitXmlDeclaration = true,
+                Indent = true,
+                NewLineOnAttributes = false,
+                Async = true
+            };
+
+            using (var xmlWriter = GetXmlWriter(settings))
```

Comment:
Should we do NuGet's thing: https://github.com/NuGet/NuGet.Client/commit/27645cbb0be2c84dd047c8287338920bd273688f?

Reply:
I think we don't need this, given that nuget does so to avoid conflicts between multiple restores and from processes trying to read the file while its being written, neither of we expect